### PR TITLE
Added three new metrics in mysqlreceiver

### PR DESCRIPTION
**Pull request by samiura**
**Description:** Added three new metrics in MySQLreceiver

**Link to tracking Issue:** #29545

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/mysqlreceiver/metadata.yaml [hunk 1]
```diff
@@ -552,6 +552,33 @@ metrics:
...
+      monotonic: true
+      aggregation_temporality: cumulative
+  mysql.qcache.queries:
+    enabled: false
+    description: The number of queries registered in the query cache.
+    unit: 1
+    sum:
+      value_type: int
+      input_type: string
+      monotonic: true
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:580]: This sounds like a "current" count, which would go up and down. If not, is it a running count of the number of queries which added to the cache at some point?

File: receiver/mysqlreceiver/metadata.yaml [hunk 2]
```diff
@@ -552,6 +552,33 @@ metrics:
       monotonic: true
       aggregation_temporality: cumulative
     attributes: [cache_status]
+  mysql.qcache.hits:
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:555]: Is there a "misses" value which could be part of this metric as well? (differentiated from "hits" by an attribute)

File: receiver/mysqlreceiver/metadata.yaml [hunk 3]
```diff
@@ -552,6 +552,33 @@ metrics:
...
+    enabled: false
+    description: The number of query cache hits.
+    unit: 1
+    sum:
+      value_type: int
+      input_type: string
+      monotonic: true
+      aggregation_temporality: cumulative
+  mysql.qcache.not.cached :
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:564]: Maybe `mysql.qcache.uncacheable`?

Comment (samiura) [reply receiver/mysqlreceiver/metadata.yaml:564]: good idea. I will change it.

File: receiver/mysqlreceiver/metadata.yaml (hunk 1 above)
Comment (samiura) [reply receiver/mysqlreceiver/metadata.yaml:580]: I agree with you after giving it a second thought. I think you're correct, it is not monotonic.

File: receiver/mysqlreceiver/metadata.yaml (hunk 2 above)
Comment (samiura) [reply receiver/mysqlreceiver/metadata.yaml:555]: Unfortunately, there are no misses value part of this metric as far as I can glean from [this](https://dev.mysql.com/doc/refman/5.7/en/server-status-variables.html#statvar_Qcache_hits). And


+-------------------------+------------+
| Variable_name           | Value      |
+-------------------------+------------+
| Qcache_free_blocks      | 1          |
| Qcache_free_memory      | 1048557680 |
| Qcache_hits             | 0          |
| Qcache_inserts          | 0          |
| Qcache_lowmem_prunes    | 0          |
| Qcache_not_cached       | 1          |
| Qcache_queries_in_cache | 0          |
| Qcache_total_blocks     | 1          |
+-------------------------+------------+

File: .chloggen/main.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: mysqlreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Added `mysql.qcache.hits`, `mysql.qcache.not.cached` and `mysql.qcache.queries` metrics.
```
Comment (atoulme) [inline .chloggen/main.yaml:10]:
```suggestion
note: Added `mysql.qcache.hits`, `mysql.qcache.uncacheable` and `mysql.qcache.queries` metrics.
```

**Review by atoulme — APPROVED** [review]

**Comment by djaglowski** [comment]
Can we answer this question before proceeding further with this PR? https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29545#issuecomment-1830902354

We explicitly support only versions 8.0+, so I'd like to understand if what was proposed is applicable there.

**Review by djaglowski — CHANGES_REQUESTED** [review]
Sorry to be a naysayer but I think we should not add metrics which are unavailable after 5.7, both because this version is deprecated and because we already don't claim support for it. I recommend the sqlquery receiver as an alternative.

**Pull request closed without merging**